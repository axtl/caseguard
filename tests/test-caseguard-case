#!/bin/sh

"$TESTDIR/hghave" caseguard || exit 80
"$TESTDIR/hghave" no-icasefs || exit 80

echo '% case-sensitive filesystem test'
hg init repo2
cat << EOF >> $HGRCPATH
[extensions]
caseguard = 
EOF
cd repo2

# multiple files, diff case
echo foo > foo
echo FOO > FOO
echo Foo > Foo
echo fOO > fOO
hg add && echo '% collision in initial add'

hg st

hg add foo FOO && echo '% collision in argument list'

echo '% case-sensitive override test'
echo baz > baz
hg add baz
mv baz BAZ
hg add -o BAZ || echo '% disable collision check failure'
mv BAZ Baz
hg add -o

hg st

cd ..
rm -rf repo2