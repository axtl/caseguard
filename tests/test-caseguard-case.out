% case-sensitive filesystem test
abort: case-collision danger
? FOO
? Foo
? fOO
? foo
abort: case-collision danger
% case-sensitive override test
adding Baz
adding FOO
adding Foo
adding fOO
adding foo
A Baz
A FOO
A Foo
A fOO
A foo
! BAZ
! baz
